src/vos/vos_io.c

```diff
@@ -1355,16 +1361,24 @@ dkey_fetch(struct vos_io_context *ioc, daos_key_t *dkey)
 	struct vos_krec_df	*krec;
 	daos_handle_t		 toh = DAOS_HDL_INVAL;
 	int			 i, rc;
+	int                      flags = 0;
 	bool			 has_cond;
 	bool			 standalone = ioc->ic_cont->vc_pool->vp_sysdb;
 
 	rc = obj_tree_init(obj);
 	if (rc != 0)
 		return rc;
 
-	rc = key_tree_prepare(obj, obj->obj_toh, VOS_BTR_DKEY,
-			      dkey, 0, DAOS_INTENT_DEFAULT, &krec,
-			      &toh, ioc->ic_ts_set);
+	if (vos_obj_flat_kv_supported(ioc->ic_cont, obj->obj_id)) {
+		if (ioc->ic_iod_nr == 1) {
```

Comment:
same here.

Reply:
Hmm, for fetch case, I guess we don't prevent multiple iod.  I should probably fix this then.